rules/android/android_binary.bzl

```diff
@@ -96,6 +99,20 @@ def android_binary(
             lint_baseline = lint_baseline,
         )
 
+    if (detekt_options.get("enabled", False) and len(attrs.get("srcs", default = [])) > 0):
+        detekt(
```

Comment:
The approach seems good and we used for lint already but I wonder if it makes sense to provide a wrapper macro that returns a `dict` instead and `kwargs` it here.  The advantage would be the paramaters would be documented. 
This pattern is already followed by other rules like `rules_jvm_external`'s [artifact](https://github.com/bazelbuild/rules_jvm_external/blob/a874c278f0bad26a86eeb682297bd38f4ed2f1e4/private/rules/artifact.bzl#L4) and our own `custom_res` macro.

Let me explain below.

Instead of 

```python

android_library(
    name = "foo"
    detekt_options = {
        "enabled": True,
        "baseline": "detekt_baseline.xml",
    },
)
```

We can create a `detect_options` macro that returns a `dict` so it becomes

```python
android_library(
    name = "foo"
    detekt_options = detect_options(
        enabled = True,
        baseline = "detekt_baseline.xml",
    ),
)
```
The advantage being since it is a macro, unknown attrs would be rejected automatically and params will be documented.

Reply:
added